test/bench/palindrome.mo

```diff
@@ -0,0 +1,62 @@
+// The 2 palindrome implementations from
+// "There and Back Again", by Olivier Danvy and Mayer Goldberg
+//
+import { error; performanceCounter; rts_heap_size; debugPrint } = "mo:⛔";
+
+actor Palindrome {
+    type List<A> = ?(A, List<A>);
+
+    func cps(xs : List<Char>) : Bool {
+        func walk(xs1 : List<Char>, xs2 : List<Char>, k : List<Char> -> Bool) : Bool =
+            switch (xs1, xs2) {
+                 case (_, null) { k xs1 };
+                 case (?(_, xs1), ?(_, null)) { k xs1 };
+                 case (?(x, xs1), ?(_, ?(_, xs2))) {
+                     walk (xs1, xs2, func (?(y, ys)) = x == y and k ys)
+                 }
+             };
+        walk (xs, xs, func null = true)
+    };
+
+    func direct(xs : List<Char>) : Bool {
+        func walk (xs1 : List<Char>, xs2 : List<Char>) : List<Char> =
+            switch (xs1, xs2) {
+                case (_, null) xs1; // even length
+                case (?(_, xs1), ?(_, null)) xs1; // odd length
+                case (?(x, xs1), ?(_, ?(_, xs2))) {
+                    let ?(y, ys) = walk (xs1, xs2);
+                    if (x == y) {
+                        ys
+                    } else { throw error("Nope") }
```

Comment:
@crusso I get error messages from here, no idea why...

Reply:
You can't try or throw from a non async context.

Reply:
Rewritten to use `do ?` instead.